packages/expo-permissions/src/Permissions.ts

```diff
@@ -30,13 +32,47 @@ export const REMINDERS = 'reminders';
 export const SYSTEM_BRIGHTNESS = 'systemBrightness';
 
 export async function getAsync(...types: PermissionType[]): Promise<PermissionResponse> {
+  if (Platform.OS === 'ios') {
+    return await _handleMultiPermissionsRequestIOSAsync(types, Permissions.getAsync);
+  }
   return await _handlePermissionsRequestAsync(types, Permissions.getAsync);
 }
 
 export async function askAsync(...types: PermissionType[]): Promise<PermissionResponse> {
+  if (Platform.OS === 'ios') {
+    return await _handleMultiPermissionsRequestIOSAsync(types, Permissions.askAsync);
+  }
   return await _handlePermissionsRequestAsync(types, Permissions.askAsync);
 }
 
+async function _handleSinglePermissionRequestIOSAsync(
+  type: PermissionType,
+  handlePermission: (type: PermissionType) => Promise<PermissionInfo>
+): Promise<PermissionInfo> {
+  return await handlePermission(type);
+}
+
+async function _handleMultiPermissionsRequestIOSAsync(
+  types: PermissionType[],
+  handlePermission: (type: PermissionType) => Promise<PermissionInfo>
+): Promise<PermissionResponse> {
+  if (!types.length) {
+    throw new Error('At least one permission type must be specified');
+  }
+
+  let permissions = {};
+  for (let type of types) {
+    permissions[type] = await _handleSinglePermissionRequestIOSAsync(type, handlePermission);
```

Comment:
what is the reason for moving the multi-permissions requests on iOS to the JS side? this is less efficient because it requires asynchronous calls over the bridge for each permission in the request rather than just one for the entire request.

Reply:
I moved it to JS side because I thought, it'll be easier to maintain this in the future. Native iOS API wasn't created to request for multiple permissions at once. That's why our code was a little bit messy and difficult to read. 
Moreover, asking/getting `permissions` isn't something that user doing very often. Mostly, it's happed once per `permission`. So I think, it wouldn't affect performance a lot.

Reply:
OK, that makes sense. Thanks for explaining.